### PR TITLE
File caching with example

### PR DESCRIPTION
**Pull request by jakedetels**
Introduced a new FileCache class to enable file-based caching.  Given FileCache is a node.js specific module (and cannot be run in the browser), I expose access to it via:

```ts
import { FileCache } from "modelfusion/server";
```

modelfusion/server re-exports everything from modelfusion/index.ts, and additionally includes any node.js specific tools.  Without bundling the code this way, we encounter a build error when attempting to import modelfusion into browser contexts (as in the nextjs example app).

### CODE REVIEW

**Comment by lgrammel** [comment]
Thanks for the pull request! I'm still wondering how to best integrate node.js specific things into the library, there are e.g. `async_hooks` and the `ws` library already. Can you for now rename "server" to "node"?

**Review by lgrammel — APPROVED** [review]

**Comment by jakedetels** [comment]
I spent considerable time trying to find an alternate way to incorporate node.js specific code, including examining the example with getRun.ts and its use of conditional imports with async_hooks.  Conditional imports wouldn't work in this case.  All the research I found indicated the best strategy is splitting the build into browser vs. node specific files.  Glad to try other options if any can be identified.

File: packages/modelfusion/src/node/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export * from "../index";
```
Comment (lgrammel) [inline packages/modelfusion/src/node/index.ts:1]: quick question: is this needed?

**Comment by lgrammel** [comment]
> I spent considerable time trying to find an alternate way to incorporate node.js specific code, including examining the example with getRun.ts and its use of conditional imports with async_hooks. Conditional imports wouldn't work in this case. All the research I found indicated the best strategy is splitting the build into browser vs. node specific files. Glad to try other options if any can be identified.

Makes sense, this is a good starting point and might be the best solution. Is the re-export of `../index` needed?

**Comment by jakedetels** [comment]
I was on the fence about the re-export of `../index`.  I've removed it.

**Comment by lgrammel** [comment]
Awesome, merged. Thank you!

**Pull request merged**